Src/ILGPU/Runtime/OpenCL/CLAccelerator.cs

```diff
@@ -101,6 +101,8 @@ internal CLAccelerator(Context context, CLDevice description)
             : base(context, description)
         {
             Backends.Backend.EnsureRunningOnNativePlatform();
+            if (!description.Capabilities.GenericAddressSpace)
```

Comment:
To my mind, it seems to be ok to throw an exception in the constructor, because the context cannot be used properly anyway without support for generic address spaces. Also, users can still output all device-specific information via the new Device API.